doc/man/tapconvert.1
.TH TAPCONVERT "1" "July 2001" "" ""
.SH NAME
tapconvert \- convert Twisted configurations from one format to another
.SH SYNOPSIS
.B tapconvert -i \fIinput\fR -o \fIoutput\fR  [-f \fIinput-type\fR] [-t \fIoutput-type\fR] [-d] [-e]
.B tapconvert --help \fIapptype\fR
.SH DESCRIPTION
.PP
The \fB\--help\fR prints out a usage message to standard output.
.TP 
\fB\--in\fR, \fB\-i\fR \fI<input file>\fR
The name of the input configuration.
.TP 
\fB\--out\fR, \fB\-o\fR \fI<output file>\fR
The name of the output configuration.
.TP 
\fB\--typein\fR, \fB\-f\fR \fI<input type>\fR
The type of the input file. Can be either 'guess', 'python', 'pickle', 'xml', or  'source'. Default is 'guess'.
.TP 
\fB\--typeout\fR, \fB\-t\fR \fI<output type>\fR
The type of the output file. Can be either 'python', 'pickle', 'xml', or  'source'. Default is 'source'.
.TP 
\fB\--decrypt\fR, \fB\-d\fR
Decrypt input.
.TP 
\fB\--encrypt\fR, \fB\-e\fR
Encrypt output.
.SH AUTHOR
Written by Moshe Zadka, based on tapconvert's help messages
.SH "REPORTING BUGS"
Report bugs to <twisted-python@twistedmatrix.com>.
.SH COPYRIGHT
Copyright \(co 2000 Matthew W. Lefkowitz
.br
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
.SH "SEE ALSO"
mktap(1)


NEWS
Ticket numbers in this file can be looked up by visiting
http://twistedmatrix.com/trac/ticket/<number>

Twisted Core 16.0.0pre2 (2016-02-28)
====================================

Features
--------
 - twisted.cred.strports has been ported to Python 3. (#8216)

Deprecations and Removals
-------------------------
 - The __version__ attribute of former subprojects (conch, mail,
   names, news, pair, runner, web, and words) is deprecated in
   preference to the central twisted.__version__. (#8219)

Other
-----
 - #8220


Twisted Conch 16.0.0pre2 (2016-02-28)
=====================================

No significant changes have been made for this release.


Twisted Mail 16.0.0pre2 (2016-02-28)
====================================

No significant changes have been made for this release.


Twisted Names 16.0.0pre2 (2016-02-28)
=====================================

No significant changes have been made for this release.


Twisted News 16.0.0pre2 (2016-02-28)
====================================

No significant changes have been made for this release.


Twisted Pair 16.0.0pre2 (2016-02-28)
====================================

No significant changes have been made for this release.


Twisted Runner 16.0.0pre2 (2016-02-28)
======================================

No significant changes have been made for this release.


Twisted Web 16.0.0pre2 (2016-02-28)
===================================

No significant changes have been made for this release.


Twisted Words 16.0.0pre2 (2016-02-28)
=====================================

No significant changes have been made for this release.


Twisted Core 16.0.0pre1 (2016-02-16)
====================================

Features
--------
 - todo parameter for IReporter.addExpectedSuccess and
   IReporter.addUnexpectedSuccess is no longer required. If not
   provided, a sensible default will be used instead. (#4811)
 - A new string endpoint type, "tls:", allows for properly-verified
   TLS (unlike "ssl:", always matching hostname resolution with
   certificate hostname verification) with faster IPv4/IPv6
   connections.  This comes with an accompanying function,
   twisted.internet.endpoints.wrapClientTLS, which can wrap an
   arbitrary client endpoint with client TLS. (#5642)
 - twisted.python.filepath.makedirs accepts an ignoreExistingDirectory
   flag which ignore the OSError raised by os.makedirs if requested
   directory already exists. (#5704)
 - twisted.protocols.amp has been ported to Python 3. (#6833)
 - twisted.internet.ssl.trustRootFromCertificates returns an object
   suitable for use as trustRoot= to
   twisted.internet.ssl.optionsForClientTLS that trusts multiple
   certificates. (#7671)
 - twisted.python.roots is now ported to Python 3. (#8131)

Bugfixes
--------
 - Expected failures from standard library unittest no longer fail
   with Trial reporters. (#4811)
 - twisted.internet.endpoints.HostnameEndpoint.connect no longer fails
   with an AlreadyCalledError when the Deferred it returns is
   cancelled after all outgoing connection attempts have been made but
   none have yet succeeded or failed. (#8014)
 - twisted.internet.task.LoopingCall.withCount when run with internal
   of 0, now calls the countCallable with 1, regardless of the time
   passed between calls. (#8125)

Improved Documentation
----------------------
 - The Twisted Project has adopted the Contributor Covenant as its
   Code of Conduct. (#8173)

Deprecations and Removals
-------------------------
 - twisted.internet.task.LoopingCall.deferred is now deprecated. Use
   the deferred returned by twisted.internet.task.LoopingCall.start()
   (#8116)
 - twisted.internet.gtkreactor, the GTK+ 1 reactor deprecated since
   Twisted 10.1, has been removed. This does not affect the GTK2,
   GLib, GTK3, or GObject-Introspection reactors. (#8145)
 - twisted.protocols.mice, containing a Logitech MouseMan serial
   driver, has been deprecated. (#8148)

Other
-----
 - #6842, #6978, #7668, #7791, #7881, #7943, #7944, #8050, #8104,
   #8115, #8119, #8122, #8139, #8144, #8154, #8162, #8180, #8187


Twisted Conch 16.0.0pre1 (2016-02-16)
=====================================

Features
--------
 - twisted.conch now uses cryptography instead of PyCrypto for its
   underlying crypto operations. (#7413)
 - twisted.conch.ssh.keys is now ported to Python 3. (#7998)

Bugfixes
--------
 - twisted.conch.ssh.channel.SSHChannel's getPeer and getHost methods
   now return an object which provides IAddress instead of an old-
   style tuple address. (#5999)
 - twisted.conch.endpoint.SSHCommandClientEndpoint, when
   authentication is delegated to an SSH agent, no longer leaves the
   agent connection opened when connection to the server is lost.
   (#8138)

Other
-----
 - #7037, #7715, #8200, #8208


Twisted Mail 16.0.0pre1 (2016-02-16)
====================================

No significant changes have been made for this release.


Twisted Names 16.0.0pre1 (2016-02-16)
=====================================

No significant changes have been made for this release.


Twisted News 16.0.0pre1 (2016-02-16)
====================================

No significant changes have been made for this release.


Twisted Pair 16.0.0pre1 (2016-02-16)
====================================

No significant changes have been made for this release.


Twisted Runner 16.0.0pre1 (2016-02-16)
======================================

No significant changes have been made for this release.


Twisted Web 16.0.0pre1 (2016-02-16)
===================================

Features
--------
 - twisted.web.http_headers._DictHeaders now correctly handles
   updating via keyword arguments in Python 3 (therefore
   twisted.web.http_headers is now fully ported to Python 3). (#6082)
 - twisted.web.wsgi has been ported to Python 3. (#7993)
 - twisted.web.http_headers.Headers now accepts both Unicode and
   bytestring keys and values, encoding to iso-8859-1 and utf8
   respectively. (#8129)
 - twisted.web.vhost ported to Python 3. (#8132)

Bugfixes
--------

trunk/twisted/scripts/tapconvert.py
# Copyright (c) Twisted Matrix Laboratories.
# See LICENSE for details.

import sys, getpass

from twisted.python import usage
from twisted.application import app
from twisted.persisted import sob


class ConvertOptions(usage.Options):
    synopsis = "Usage: tapconvert [options]"
    optParameters = [
        ['in',      'i', None,     "The filename of the tap to read from"],
        ['out',     'o', None,     "A filename to write the tap to"],
        ['typein',  'f', 'guess',
         "The  format to use; this can be 'guess', 'python', "
         "'pickle', 'xml', or 'source'."],
        ['typeout', 't', 'source',
         "The output format to use; this can be 'pickle', 'xml', or 'source'."],
        ]

    optFlags = [
        ['decrypt', 'd', "The specified tap/aos/xml file is encrypted."],
        ['encrypt', 'e', "Encrypt file before writing"]
        ]

    compData = usage.Completions(
        optActions={"typein": usage.CompleteList(["guess", "python", "pickle",
                                                  "xml", "source"]),
                    "typeout": usage.CompleteList(["pickle", "xml", "source"]),
                    "in": usage.CompleteFiles(descr="tap file to read from"),
                    "out": usage.CompleteFiles(descr="tap file to write to"),
                    }
        )

    def postOptions(self):
        if self['in'] is None:
            raise usage.UsageError("%s\nYou must specify the input filename."
                                   % self)
        if self["typein"] == "guess":
            try:
                self["typein"] = sob.guessType(self["in"])
            except KeyError:
                raise usage.UsageError("Could not guess type for '%s'" %
                                       self["typein"])

def run():
    options = ConvertOptions()
    try:
        options.parseOptions(sys.argv[1:])
    except usage.UsageError, e:
        print e
    else:
        app.convertStyle(options["in"], options["typein"],
                     options.opts['decrypt'] or getpass.getpass('Passphrase: '),
                     options["out"], options['typeout'], options["encrypt"])
